### PR TITLE
Add a healthcheck to postgres

### PR DESCRIPTION
**Pull request by ajm188**
The relevant step is the test:prepare and spec:prepare tasks. We
"enhance" them (rake-speak) to do the custom wait_for_db task. This
depends on environment so that the database config is loaded.

Note that we only do this "enhancement" if we're in docker, which we
detect through the use of an environment variable which is set in the
compose file.

### CODE REVIEW

**Comment by ajm188** [comment]
Hmm I'm not convinced though. I wish I could come up with a way to more reliably repro

**Comment by ajm188** [comment]
Got a repro. The "database does not exist" is a docker cache issue. The `rake db:test:prepare` CMD inside the container is what creates the database, so on a second build, docker views this layer as cached, so the command doesn't actually get run against the new db container.

So, moving the cmd out of the dockerfile and explicitly running it in the make target is a step in the right direction. I think then that we can remove the `ActiveRecord::NoDatabaseError` from our `rescue` block in `connect_to_db`

**Comment by ajm188** [comment]
@matthewbentley this is ready for review. Thoughts? I'm not sure we should loop forever, and rely on travis to kill us after 10 minutes, since locally this would require a user to ctrl-C the build.

File: Rakefile [hunk 1]
```diff
@@ -4,3 +4,29 @@
 require File.expand_path('../config/application', __FILE__)
 
 Rails.application.load_tasks
+
+if ENV['IN_DOCKER']
+  def wait_for_db_container
+    loop do
+      return if connect_to_db
+      sleep 1
```
Comment (matthewbentley) [inline Rakefile]: I think you could add a counter here, and if you hit a time (5 minutes?) raise an exception.  I'd assume that makes the tests fail.

Comment (ajm188) [reply Rakefile]: Yeah travis kills us after 10 minutes. Up to you whether we should continue to just let that happen.

Comment (matthewbentley) [reply Rakefile]: I'm not a huge fan of requiring users to ctrl+c after some amount of time, I'd rather an error that says what the problem probably is (ie can't connect to postgres).

Comment (ajm188) [reply Rakefile]: +1, updated

**Pull request merged**